articles/golang-advanced-programming-concepts/index.md

```diff
@@ -0,0 +1,174 @@
+
+import "fmt"
+
+func main() {
+	var test[10] int
+	// assigning the value 10 to the first index in the array
+	test[0] = 10
+        // assigning the value 20 to the second index
+        test[1] = 20
+	// printing the first element
+	fmt.Println(test[0])
+	
+	var test2[1] string
+	test2[0] = "hello"
+	fmt.Println(test2[0])
+}
+```
+In the above example, we create an array of integers called "test" with a length of 10. We then assign the values 10 and 20 to the first and second index, respectively. We can retrieve and print the first value by referring to the index 0. 
+
+```go
+package main
+
+import "fmt"
+
+func main() {
+	var test[10] int
+	for i := 0; i < 10; i++{
+		test[i] = i
+	}
+	
+	for i := 0; i < 10; i++{
+		fmt.Println(test[i])
+	}
+}
+```
+
+We can also use loops to assign values to the array. In the above example, we assign values from 0 to 9 to the elements in the array and then print them on the screen.
+
+### Structures
+A [structure](https://medium.com/rungo/structures-in-go-76377cc106a2), also called struct, is similar to a [class](https://en.wikipedia.org/wiki/Class_(computer_programming)) in the [object-oriented](https://searchapparchitecture.techtarget.com/definition/object-oriented-programming-OOP) paradigm. For those that are not familiar with OOP or object-oriented programming, structs are user-defined types that are used to group similar fields of the same or different data types together. For instance, details about a student such as his name, score, age, etc. are values of different data types, but it makes sense to group them together instead of creating different variables for each of them.  
+
+The syntax for creating a struct is as follows:
+
+```
+type <name of the structure> struct {
+    field1 <data type>
+    field2 <data type>
+}
+```
+Example:
+
+```
+package main
+
+import "fmt"
+
+type Student struct {
+	name string
+	age int
+	score float64
+}
+
+func main() {
+    var s1 = Student{"Adith", 1, 99.5}	
+    fmt.Println(s1)
+    fmt.Println(s1.name)
+    fmt.Println(s1.age)
+    fmt.Println(s1.score)
+	
+    var s2 = Student{}
+    s2.name = "Bharadwaj"
+    fmt.Println(s2.name)
+}
+```
+In the above example, we create a struct called "Student" with three fields: `name`, `age`, and `score` of types `string`, `int`, and `float64`, respectively. In the main function, we create a variable called `s1` of type `Student` and initialize it with values that correspond to the `name`, `age`, and `score`. The curly braces({}) are used to initialize the values of a struct. 
+
+The `dot(.)` operator can also be used to assign values to the fields in the struct. In the example above, we create another variable called `s2` of type `Student` without initializing it. We then use the `dot(.)` operator to assign the value "Bharadwaj" to the name.
+
+### Putting it all together
+We are going to write a program to create a `struct` called "Student", get the details of 5 students, and display them to the user.
+
+```go
```

Comment:
This code snippet is well written, but it might help to add some comments to aid the reader to keep track 😅